### PR TITLE
Allow array values for attributes

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Resolves https://github.com/open-telemetry/opentelemetry-specification/issues/341

Array values are useful for representing attributes that can have
multiple values (e.g. representing a Memcached `get` that can be
over multiple keys and making the keys a span attribute).

This change allows homogeneous array values for span attributes
and specifies that array values should be JSON encoded string when
exporting using protocols that do not natively support array values.

### CODE REVIEW

File: specification/api-tracing.md [hunk 1]
```diff
@@ -344,8 +344,10 @@ A `Span` MUST have the ability to set attributes associated with it.
 An `Attribute` is defined by the following properties:
 
 - (Required) The attribute key, which must be a string.
-- (Required) The attribute value, which must be either a string, a boolean
-  value, or a numeric type.
+- (Required) The attribute value, which is either:
+  - A primitive type: string, boolean or numeric.
+  - An array of primitive type values. The array MUST be homogeneous,
```
Comment (bogdandrutu) [inline specification/api-tracing.md]: I think the requirement is not to support array, but a map so that you have {"http" : {"url":...,...}}

Comment (tigrannajaryan) [reply specification/api-tracing.md]: That is a different requirement. We would need that for https://github.com/open-telemetry/opentelemetry-specification/issues/76

This one is only adding support for arrays which is needed by https://github.com/open-telemetry/opentelemetry-specification/issues/341

Comment (bogdandrutu) [reply specification/api-tracing.md]: Before accepting this, I am curios to see all the requirements. Is only map/array that we need?

Comment (tigrannajaryan) [reply specification/api-tracing.md]: The only requirement we have so far is in https://github.com/open-telemetry/opentelemetry-specification/issues/341 and it is only for arrays, not maps.

Comment (bogdandrutu) [reply specification/api-tracing.md]: I remember for sure that I saw the map requirement documented as well, for the example I mentioned in the previous post.

Comment (jmacd) [reply specification/api-tracing.md]: To complete the JSON structure, we'd allow a `null` value in addition to the ones already discussed. There may be some connection to the discussion about unspecified values for metric labels.

(For the record, I don't think that metric labels need to support lists or maps of values--but I also don't see a problem with coercing JSON lists or maps to strings when used as metric labels. Supporting a null value distinct from the string "null" or empty string would let provide a satisfying answer to the question about unspecified values--whereas the current state was not completely satisfying.)

Comment (tigrannajaryan) [reply specification/api-tracing.md]: This is only for Span and Resource attributes. It does not affect metric labels.

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan as a followup from the discussions in this thread I think we need to file an issue to discuss the support of the maps. @Oberon00 @arminru I think this idea of supporting a map was discuss during some discussions about typed spans.

Added a an issue to discuss: https://github.com/open-telemetry/opentelemetry-specification/issues/376

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers can we merge this?

**Pull request merged**